### DISCUSSION TITLE
Trying to bypass auth for specific url

### DISCUSSION THREAD

**Discussion by Draghmar** [OpenIDC/mod_auth_openidc#504] [Q&A]
Hi
I have mod_auth_openidc setup to auth with Google for one site. But now I'm in need to give access to one url and struggle to find solution.
Let's say the site `https://example.com` is protected and `https://example.com/api` is the one needs to be bypassed. I tried to make api folder exist, which didn't make it work, as well as virtual one, where it was rewritten to the php file - this didn't work either.
I did found [this](https://github.com/zmartzone/mod_auth_openidc/issues/324) solution but for some unknown reason it doesn't work for me. Maybe it is somehow related to some issue I have where I can't get Auth to work when set directly in `VirtualHost->Location`. I have to set in `.httaccess` instead.

So I tried few things that was mostly related to move all the options around. The base options currently are:
```
<Location /www/example.com/>
AuthType openid-connect
Require claim hd:---
</Location>
<Location /www/example.com/api/>
Satisfy Any
Allow from all
</Location>
```
And in .htaccess I have:
```
AuthType openid-connect
Require claim hd:---
```
Removing `.htaccess` entry basically disables authorization like it weren't set in `Location` at all.
Removing Location entries and set additional `.htaccess` in `/api` folder doesn't work either - I still get prompted for auth whether it's `https://example.com` or `https://example.com/api`.
I can change .htaccess to:
```
AuthType openid-connect
<RequireAny>
Require ip ---
Require claim hd:---
</RequireAny>
```
and it will work properly by passing specified IP and block others. But that is not what I'd like to do here. :P

Any suggestions?

**Comment by zandbelt** [comment]
you'll need to use `AuthType None` in locations you want to bypass

**Comment by Draghmar** [comment]
You mean location or Location? :P And to make sure, `/api` doesn't exists. It's simply rewritten to php file.
Anyway I tried that already, both of them. And it didn't work. I'm guessing this has something to do with the fact that settings from `VirtualHost` aren't taken into consideration for some unknown reason. Because in order for the one in `VirtualHost` to work I have to remove one in `.htaccess` but then mod_auth doesn't work at all. Like it wasn't setup at all. 
Maybe I should actually start discussion with that in first place?
